packages/@aws-cdk/pipelines/lib/actions/publish-assets-action.ts

```diff
@@ -138,15 +138,6 @@ export class PublishAssetsAction extends CoreConstruct implements codepipeline.I
       role: props.role,
     });
 
-    const rolePattern = props.assetType === AssetType.DOCKER_IMAGE
-      ? 'arn:*:iam::*:role/*-image-publishing-role-*'
-      : 'arn:*:iam::*:role/*-file-publishing-role-*';
-
-    project.addToRolePolicy(new iam.PolicyStatement({
-      actions: ['sts:AssumeRole'],
-      resources: [rolePattern],
-    }));
```

Comment:
Worth noting that this entire section was a (thankfully redundant) no-op. The role passed in is an ImmutableRole, and so this section never really had any impact.